app/views/courses/TeacherClassView.coffee

```diff
@@ -0,0 +1,233 @@
+    @$('#join-url-input').val(@joinURL).select()
+    @tryCopy()
+    
+  tryCopy: ->
+    try
+      document.execCommand('copy')
+      application.tracker?.trackEvent 'Classroom copy URL', category: 'Courses', classroomID: @classroom.id, url: @joinURL
+    catch err
+      message = 'Oops, unable to copy'
+      noty text: message, layout: 'topCenter', type: 'error', killer: false
+
+  onClickAddStudents: (e) =>
+    modal = new InviteToClassroomModal({ classroom: @classroom })
+    @openModalView(modal)
+    @listenToOnce modal, 'hide', @render
+    
+  sortByName: (e) ->
+    if @sortValue == 'name'
+      @sortDirection = -@sortDirection
+    else
+      @sortValue = 'name'
+      @sortDirection = 1
+      
+    dir = @sortDirection
+    @students.comparator = (student1, student2) ->
+      return (if student1.get('name') < student2.get('name') then -dir else dir)
+    @students.sort()
+    
+  sortByProgress: (e) ->
+    if @sortValue == 'progress'
+      @sortDirection = -@sortDirection
+    else
+      @sortValue = 'progress'
+      @sortDirection = 1
+      
+    dir = @sortDirection
+    
+    @students.comparator = (student) ->
+      #TODO: I would like for this to be in the Level model,
+      #      but it doesn't know about its own courseNumber
+      level = student.latestCompleteLevel
+      if not level
+        return -dir
+      return dir * ((1000 * level.courseNumber) + level.levelNumber)
+    @students.sort()
+  
+  getSelectedStudentIDs: ->
+    $('.student-row .checkbox-flat input:checked').map (index, checkbox) ->
+      $(checkbox).data('student-id')
+    
+  ensureInstance: (courseID) ->
+    
+  onClickEnroll: (e) ->
+    userID = $(e.currentTarget).data('user-id')
+    user = @students.get(userID)
+    selectedUsers = new Users([user])
+    modal = new ActivateLicensesModal { @classroom, selectedUsers, users: @students }
+    @openModalView(modal)
+    modal.once 'redeem-users', -> document.location.reload()
+    application.tracker?.trackEvent 'Classroom started enroll students', category: 'Courses'
+  
+  onClickBulkEnroll: ->
+    courseID = $('.bulk-course-select').val()
+    courseInstance = @courseInstances.findWhere({ courseID, classroomID: @classroom.id })
+    userIDs = @getSelectedStudentIDs().toArray()
+    selectedUsers = new Users(@students.get(userID) for userID in userIDs)
+    modal = new ActivateLicensesModal { @classroom, selectedUsers, users: @students }
+    @openModalView(modal)
+    modal.once 'redeem-users', -> document.location.reload()
+    application.tracker?.trackEvent 'Classroom started enroll students', category: 'Courses'
+    
+  onClickBulkAssign: ->
+    courseID = $('.bulk-course-select').val()
+    courseInstance = @courseInstances.findWhere({ courseID, classroomID: @classroom.id })
+    members = @getSelectedStudentIDs().toArray()
+    if courseInstance
+      for studentID in members
+        courseInstance.addMember studentID, {
+          success: =>
+            @render()
```

Comment:
This sort of thing will throw an error if the user navigates away and destroys the view before the network call finishes. However, it's just a JS error, so they won't notice, but our error-tracking tools would, and we would if we were testing or they were an admin. I tend to do `@render() unless @destroyed` in cases like this when I remember (see [the generic destructor](https://github.com/codecombat/codecombat/blob/master/app/core/CocoClass.coffee#L37-L47)).